packages/txservice/src/chainreader.ts

```diff
@@ -352,17 +349,16 @@ export class ChainReader {
    * @param whichChain - Whether it's sender or receiver chain, used for
    * logging purposes only.
    */
-  private async calculateGasFee(
+  public async calculateGasFee(
     chainId: number,
     gasPriceChainId: number,
     assetId: string,
     nativeTokenChainId: number,
     nativeTokenAssetId: string,
     decimals: number,
-    method: "prepare" | "fulfill",
+    method: "prepare" | "fulfill" | "cancel",
```

Comment:
needs to also have a flag to know if this is a router contract or direct transaction manager prepare, and should have different estimates for both

Reply:
right! we have flag in getContext() called isRouterContract. Need to add those estimates

Reply:
updated